engine/src/main/java/com/datatorrent/stram/codec/LogicalPlanSerializer.java

```diff
@@ -116,91 +172,110 @@ public boolean useForType(JavaType t)
+  {
+
+    HashMap<String, Object> streamDetailMap = new HashMap<String, Object>();
+    String streamName = streamMeta.getName();
+    String sourcePortName = streamMeta.getSource().getPortName();
+    OperatorMeta operatorMeta = streamMeta.getSource().getOperatorMeta();
+    HashMap<String, Object> sourcePortDetailMap = new HashMap<String, Object>();
+    sourcePortDetailMap.put("operatorName", operatorMeta.getName());
+    sourcePortDetailMap.put("portName", sourcePortName);
+    streamDetailMap.put("name", streamName);
+    streamDetailMap.put("source", sourcePortDetailMap);
+    List<InputPortMeta> sinks = streamMeta.getSinks();
+    ArrayList<HashMap<String, Object>> sinkPortList = new ArrayList<HashMap<String, Object>>();
+    for (InputPortMeta sinkPort : sinks) {
+      HashMap<String, Object> sinkPortDetailMap = new HashMap<String, Object>();
+      sinkPortDetailMap.put("operatorName", sinkPort.getOperatorWrapper().getName());
+      sinkPortDetailMap.put("portName", sinkPort.getPortName());
+      sinkPortList.add(sinkPortDetailMap);
+    }
+    streamDetailMap.put("sinks", sinkPortList);
+    if (streamMeta.getLocality() != null) {
+      streamDetailMap.put("locality", streamMeta.getLocality().name());
+    }
+    return streamDetailMap;
+  }
+
+  /**
+   *
+   * Return information about module and sub-dag populated by it.
+   *
+   * @param dag top level DAG
+   * @param moduleMeta module information. DAG within module is used for constructing response.
+   * @return
+   */
+  private static Map<String, Object> getLogicalModuleDetails(LogicalPlan dag, ModuleMeta moduleMeta)
+  {
+    Map<String, Object> moduleDetailMap = new HashMap<String, Object>();
+    ArrayList<Map<String, Object>> operatorArray = new ArrayList<Map<String, Object>>();
+    ArrayList<Map<String, Object>> streamArray = new ArrayList<Map<String, Object>>();
+    moduleDetailMap.put("name", moduleMeta.getName());
+    moduleDetailMap.put("className", moduleMeta.getModule().getClass().getName());
+
+    moduleDetailMap.put("operators", operatorArray);
+    for (OperatorMeta operatorMeta : moduleMeta.getDag().getAllOperators()) {
+      if (operatorMeta.getModuleName() == null) {
+        String fullName = moduleMeta.getFullName() + LogicalPlan.MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
+        operatorArray.add(getLogicalOperatorDetails(dag.getOperatorMeta(fullName)));
       }
-      streamDetailMap.put("sinks", sinkPortList);
-      if (streamMeta.getLocality() != null) {
-        streamDetailMap.put("locality", streamMeta.getLocality().name());
+    }
+    moduleDetailMap.put("streams", streamArray);
+    for (StreamMeta sMeta : moduleMeta.getDag().getAllStreams()) {
+      if (sMeta.getModuleName() == null) {
+        String fullName = moduleMeta.getFullName() + LogicalPlan.MODULE_NAMESPACE_SEPARATOR + sMeta.getName();
+        streamArray.add(getLogicalStreamDetails(dag.getStream(fullName)));
       }
     }
-    return result;
```

Comment:
Also made changes to return only name of the operator and expanded info for sub modules, the flattened dag response looks like as below.

```json
"modules":[
   {
      "name":"Me",
      "className":"com.datatorrent.modules.demo.Level3Module",
      "operators":[
         "Me$O1"
      ],
      "modules":[
         {
            "name":"M1",
            "className":"com.datatorrent.modules.demo.Level2ModuleB",
            "operators":[
               "Me$M1$O2",
               "Me$M1$O1"
            ],
            "modules":[
               {
                  "name":"M1",
                  "className":"com.datatorrent.modules.demo.Level1Module",
                  "operators":[
                     "Me$M1$M1$O1"
                  ],
                  "modules":[
                  ]
               }
            ]
         },
         {
            "name":"M2",
            "className":"com.datatorrent.modules.demo.Level1Module",
            "operators":[
               "Me$M2$O1"
            ],
            "modules":[
            ]
         }
      ]
   }
]
}
```